jbang fails to resolve comma delimited list of dependencies in jbang.properties

jbang.properties used to support run.deps property with a comma delimited list of default dependencies.

```properties
# Declare default dependencies
run.deps=org.foo:bar:1.0,org.baz:qux:2.0
```

This fails with this error:
```
[ERROR] Could not read artifact descriptor for org.foo:bar:jar:qux:2.0:1.0,org.baz
dev.jbang.cli.ExitException: Could not read artifact descriptor for org.foo:bar:jar:qux:2.0:1.0,org.baz
	at dev.jbang.dependencies.ArtifactResolver.resolveDescriptor(ArtifactResolver.java:379)
	at dev.jbang.dependencies.ArtifactResolver.getManagedDependencies(ArtifactResolver.java:348)
	at dev.jbang.dependencies.ArtifactResolver.lambda$resolve$1(ArtifactResolver.java:209)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at dev.jbang.dependencies.ArtifactResolver.resolve(ArtifactResolver.java:210)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:118)
	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:66)
	at dev.jbang.source.BuildContext.resolveClassPath(BuildContext.java:118)
	at dev.jbang.source.BuildContext.isUpToDate(BuildContext.java:106)
	at dev.jbang.source.AppBuilder.build(AppBuilder.java:62)
	at dev.jbang.source.AppBuilder.build(AppBuilder.java:22)
	at dev.jbang.cli.Run.doCall(Run.java:106)
	at dev.jbang.cli.BaseCommand.execute(BaseCommand.java:134)
	at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
	at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:239)
	at org.aesh.command.impl.AeshCommandRuntime.runExecutor(AeshCommandRuntime.java:222)
	at org.aesh.command.impl.AeshCommandRuntime.executeCommand(AeshCommandRuntime.java:175)
	at org.aesh.AeshRuntimeRunner.execute(AeshRuntimeRunner.java:160)
	at dev.jbang.Main.main(Main.java:46)
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.foo:bar:jar:qux:2.0:1.0,org.baz
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:254)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:183)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor(DefaultRepositorySystem.java:269)
	at dev.jbang.dependencies.ArtifactResolver.resolveDescriptor(ArtifactResolver.java:377)
	... 24 more
```

Seems like the complete comma delimited list of dependencies is passed to the resolver as a Maven GAV when dependencies are defined in jbang.properties.

Passing the same comma delimited list as a jbang command line argument (`--deps`) works fine

--- a/build.gradle
+++ b/build.gradle
@@ -24,7 +24,7 @@ plugins {
 def allureVersion = '2.29.1'
 def aspectJVersion = '1.9.25'
 def assertjVersion = '3.27.7'
-def aeshVersion = '3.16.4'
+def aeshVersion = '3.16.5'
 def devkitmanVersion = '0.4.9'
 def junitBomVersion = '5.14.4'
 def testcontainersVersion = '2.0.2'

--- a/src/test/java/dev/jbang/cli/TestAeshParsing.java
+++ b/src/test/java/dev/jbang/cli/TestAeshParsing.java
@@ -341,6 +341,76 @@ class TestAeshParsing extends BaseTest {
 		assertThat(run.runMixin.debugString, is(nullValue()));
 	}
 
+	// --- #2587: --runtime-option with space separator ---
+
+	@Test
+	void testRuntimeOptionWithSpaceSeparator() {
+		// --runtime-option -Dstdout.encoding=UTF-8 should work (space between option
+		// and value)
+		Run run = JBang.parseCommand("run", "--runtime-option", "-Dstdout.encoding=UTF-8", "test.java");
+		assertThat(run.runMixin.javaRuntimeOptions, hasSize(1));
+		assertThat(run.runMixin.javaRuntimeOptions, contains("-Dstdout.encoding=UTF-8"));
+	}
+
+	@Test
+	void testMultipleRuntimeOptionsWithSpaceSeparator() {
+		// Multiple --runtime-option with space should all be parsed
+		Run run = JBang.parseCommand("run",
+				"--runtime-option", "-Dstdout.encoding=UTF-8",
+				"--runtime-option", "-Dpython.console.encoding=UTF-8",
+				"test.java");
+		assertThat(run.runMixin.javaRuntimeOptions, hasSize(2));
+		assertThat(run.runMixin.javaRuntimeOptions,
+				contains("-Dstdout.encoding=UTF-8", "-Dpython.console.encoding=UTF-8"));
+	}
+
+	@Test
+	void testRuntimeOptionShortWithSpaceSeparator() {
+		// -R -Dkey=value (space between -R and value) should work
+		Run run = JBang.parseCommand("run", "-R", "-Dstdout.encoding=UTF-8", "test.java");
+		assertThat(run.runMixin.javaRuntimeOptions, hasSize(1));
+		assertThat(run.runMixin.javaRuntimeOptions, contains("-Dstdout.encoding=UTF-8"));
+	}
+
+	// --- #2589: --deps comma-separated via DefaultValueProvider ---
+
+	@Test
+	void testDepsDefaultValueProviderCommaSeparated() throws IOException {
+		// When run.deps is set in config with comma-delimited deps,
+		// they should be split into separate list items (not treated as one GAV)
+		String config = "run.deps = org.foo:bar:1.0,org.baz:qux:2.0\n";
+		Files.write(jbangTempDir.resolve(Configuration.JBANG_CONFIG_PROPS), config.getBytes());
+		Configuration.instance(null);
+
+		Run run = JBang.parseCommand("run", "test.java");
+		assertThat(run.dependencyInfoMixin.dependencies, hasSize(2));
+		assertThat(run.dependencyInfoMixin.dependencies,
+				contains("org.foo:bar:1.0", "org.baz:qux:2.0"));
+	}
+
+	@Test
+	void testDepsDefaultValueProviderSingleDep() throws IOException {
+		// Single dep from config should still work
+		String config = "run.deps = org.foo:bar:1.0\n";
+		Files.write(jbangTempDir.resolve(Configuration.JBANG_CONFIG_PROPS), config.getBytes());
+		Configuration.instance(null);
+
+		Run run = JBang.parseCommand("run", "test.java");
+		assertThat(run.dependencyInfoMixin.dependencies, hasSize(1));
+		assertThat(run.dependencyInfoMixin.dependencies, contains("org.foo:bar:1.0"));
+	}
+
+	@Test
+	void testDepsExplicitOverridesDefaultValueProvider() throws IOException {
+		// Explicit --deps should override config defaults
+		String config = "run.deps = org.foo:bar:1.0,org.baz:qux:2.0\n";
+		Files.write(jbangTempDir.resolve(Configuration.JBANG_CONFIG_PROPS), config.getBytes());
+		Configuration.instance(null);
+
+		Run run = JBang.parseCommand("run", "--deps", "org.other:lib:3.0", "test.java");
+		assertThat(run.dependencyInfoMixin.dependencies, hasItem("org.other:lib:3.0"));
+	}
+
 	// --- Subcommand list consistency ---
 
 	@Test

